lib/install.js

```diff
@@ -39,8 +39,15 @@ function install (gyp, argv, callback) {
     }
   }
 
-  var distUrl = gyp.opts['dist-url'] || gyp.opts.disturl || 'https://nodejs.org/dist'
+  var distUrl = gyp.opts['dist-url'] || gyp.opts.disturl
 
+  if (!distUrl) {
+    if (semver.satisfies(version, '>=1.0.0')) {
```

Comment:
"Node.js" 1.0.0 will never be released. As a TSC member, I can authoritatively say that. :)

Also, io.js is node.js now: https://github.com/nodejs/node

However, version detection still isn't a great way of doing things, though it could for for io.js `1.0.0> <3.0.0`. As @camsong said, what we really should be doing now is detecting 3.0.0's [`process.release`](https://iojs.org/api/process.html#process_process_release) :)